Delivered-To: koen@beagleboard.org
Received: by 10.216.36.75 with SMTP id v53cs218006wea;
        Fri, 8 Jan 2010 07:39:19 -0800 (PST)
Received: by 10.115.103.23 with SMTP id f23mr12356197wam.226.1262965158521;
        Fri, 08 Jan 2010 07:39:18 -0800 (PST)
Return-Path: <3n1FHSwYPC4Mrohzptilhnslivhyk.vynilhnslivhyknvvnslnyv1wz.jvt@listserv.bounces.google.com>
Received: from mail-pz0-f138.google.com (mail-pz0-f138.google.com [209.85.222.138])
        by mx.google.com with ESMTP id 9si7274234pzk.15.2010.01.08.07.39.16;
        Fri, 08 Jan 2010 07:39:17 -0800 (PST)
Received-SPF: pass (google.com: domain of 3n1FHSwYPC4Mrohzptilhnslivhyk.vynilhnslivhyknvvnslnyv1wz.jvt@listserv.bounces.google.com designates 209.85.222.138 as permitted sender) client-ip=209.85.222.138;
Authentication-Results: mx.google.com; spf=pass (google.com: domain of 3n1FHSwYPC4Mrohzptilhnslivhyk.vynilhnslivhyknvvnslnyv1wz.jvt@listserv.bounces.google.com designates 209.85.222.138 as permitted sender) smtp.mail=3n1FHSwYPC4Mrohzptilhnslivhyk.vynilhnslivhyknvvnslnyv1wz.jvt@listserv.bounces.google.com; dkim=pass (test mode) header.i=@googlegroups.com
Received: by pzk2 with SMTP id 2sf6049133pzk.27
        for <multiple recipients>; Fri, 08 Jan 2010 07:39:16 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=googlegroups.com; s=beta;
        h=domainkey-signature:received:x-beenthere:received:received:received
         :received:received-spf:received:mime-version:received:date
         :message-id:subject:from:to:x-original-authentication-results
         :x-original-sender:reply-to:precedence:mailing-list:list-id
         :list-post:list-help:list-archive:x-thread-url:x-message-url:sender
         :list-unsubscribe:list-subscribe:content-type;
        bh=NrT5b+hNQLcUxuuUxSE9xgP8l6ppaVDT/n2MbfizXXg=;
        b=yBQ3bq0ciS6qKoPZPcLLFORPxwQxMLQ8fCRLMIKRzXLFudf7oKso90LmSLTOdBPG1/
         VtP5SHwH94/9qMfiBt6tP/ctGmRgaiInagDDK6Vm9HPeFwlSnw3vEs+us9uS8+M0nq0r
         w4fZBAVg88EjhyYye8sR9HTxWqoD6MmoFZ8xw=
DomainKey-Signature: a=rsa-sha1; c=nofws;
        d=googlegroups.com; s=beta;
        h=x-beenthere:received-spf:mime-version:date:message-id:subject:from
         :to:x-original-authentication-results:x-original-sender:reply-to
         :precedence:mailing-list:list-id:list-post:list-help:list-archive
         :x-thread-url:x-message-url:sender:list-unsubscribe:list-subscribe
         :content-type;
        b=MC8JBr0AHWoi6cJVeRDSuaJByBiotlgA7WEQNSwY8VNKj//50+GmZE2JfItV+S56wH
         IHiW5gpkGyoK+x1iSfjl6RGBaI5S6hdKPtzI4VszemAIGI0NlxZmQAnOQPF3WQCyXcGX
         elyK18i6/LeZeK8Ct1BaANDnPfNgkbBGoStpw=
Received: by 10.114.6.29 with SMTP id 29mr2535635waf.24.1262965151011;
        Fri, 08 Jan 2010 07:39:11 -0800 (PST)
X-BeenThere: beagleboard@googlegroups.com
Received: by 10.115.116.23 with SMTP id t23ls486004wam.1.p; Fri, 08 Jan 2010 
	07:39:00 -0800 (PST)
Received: by 10.114.187.11 with SMTP id k11mr2703732waf.28.1262965139892;
        Fri, 08 Jan 2010 07:38:59 -0800 (PST)
Received: by 10.114.187.11 with SMTP id k11mr2703731waf.28.1262965139862;
        Fri, 08 Jan 2010 07:38:59 -0800 (PST)
Return-Path: <khasim@beagleboard.org>
Received: from mail-px0-f198.google.com (mail-px0-f198.google.com [209.85.216.198])
        by gmr-mx.google.com with ESMTP id 16si3405078pxi.3.2010.01.08.07.38.59;
        Fri, 08 Jan 2010 07:38:59 -0800 (PST)
Received-SPF: neutral (google.com: 209.85.216.198 is neither permitted nor denied by best guess record for domain of khasim@beagleboard.org) client-ip=209.85.216.198;
Received: by pxi36 with SMTP id 36so13875651pxi.26
        for <beagleboard@googlegroups.com>; Fri, 08 Jan 2010 07:38:59 -0800 (PST)
MIME-Version: 1.0
Received: by 10.142.250.5 with SMTP id x5mr2689870wfh.341.1262965139614; Fri, 
	08 Jan 2010 07:38:59 -0800 (PST)
Date: Fri, 8 Jan 2010 21:08:59 +0530
Message-ID: <a8ca84ad1001080738q6f5aeca1gd276ed5846b0db43@mail.gmail.com>
Subject: [beagleboard] TI:OMAP: [PATCH 2/4] Enable I2C bus switching
From: Khasim Syed Mohammed <khasim@beagleboard.org>
To: u-boot@lists.denx.de, beagleboard@googlegroups.com
X-Original-Authentication-Results: gmr-mx.google.com; spf=neutral (google.com: 
	209.85.216.198 is neither permitted nor denied by best guess record for 
	domain of khasim@beagleboard.org) smtp.mail=khasim@beagleboard.org
X-Original-Sender: khasim@beagleboard.org
Reply-To: beagleboard@googlegroups.com
Precedence: list
Mailing-list: list beagleboard@googlegroups.com; contact beagleboard+owners@googlegroups.com
List-ID: <beagleboard.googlegroups.com>
List-Post: <http://groups.google.com/group/beagleboard/post?hl=>, 
	<mailto:beagleboard@googlegroups.com>
List-Help: <http://groups.google.com/support/?hl=>, <mailto:beagleboard+help@googlegroups.com>
List-Archive: <http://groups.google.com/group/beagleboard?hl=>
X-Thread-Url: http://groups.google.com/group/beagleboard/t/806cd9676271c3e3
X-Message-Url: http://groups.google.com/group/beagleboard/msg/d4cf808abcb065f9
Sender: beagleboard@googlegroups.com
List-Unsubscribe: <http://groups.google.com/group/beagleboard/subscribe?hl=>, 
	<mailto:beagleboard+unsubscribe@googlegroups.com>
List-Subscribe: <http://groups.google.com/group/beagleboard/subscribe?hl=>, 
	<mailto:beagleboard+subscribe@googlegroups.com>
Content-Type: multipart/mixed; boundary=0016e64cc3d48ce7bb047ca8fca8

--0016e64cc3d48ce7bb047ca8fca8
Content-Type: text/plain; charset=ISO-8859-1

From 9045377f255e8a59450a6957e63366b4963281ae Mon Sep 17 00:00:00 2001
From: Syed Mohammed Khasim <khasim@ti.com>
Date: Fri, 8 Jan 2010 20:20:41 +0530
Subject: [PATCH] Enable I2C bus switching

OMAP3 supports Multiple I2C channels, this patch allows
us to use i2c dev <bus no> command to switch between busses.

Signed-off-by: Syed Mohammed Khasim <khasim@ti.com>
Acked-by: Heiko Schocher <hs@denx.de>
---
 drivers/i2c/omap24xx_i2c.c     |    5 +++++
 include/configs/omap3_beagle.h |    4 ++++
 2 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/i2c/omap24xx_i2c.c b/drivers/i2c/omap24xx_i2c.c
index ff18991..e8c8184 100644
--- a/drivers/i2c/omap24xx_i2c.c
+++ b/drivers/i2c/omap24xx_i2c.c
@@ -435,3 +435,8 @@ int i2c_set_bus_num(unsigned int bus)

 	return 0;
 }
+
+int i2c_get_bus_num(void)
+{
+	return (int) current_bus;
+}
diff --git a/include/configs/omap3_beagle.h b/include/configs/omap3_beagle.h
index d1c9cd0..ff6d432 100644
--- a/include/configs/omap3_beagle.h
+++ b/include/configs/omap3_beagle.h
@@ -100,6 +100,10 @@
 /* DDR - I use Micron DDR */
 #define CONFIG_OMAP3_MICRON_DDR		1

+/* Enable Multi Bus support for I2C */
+#define CONFIG_I2C_MULTI_BUS		1
+#define CONFIG_SYS_I2C_NOPROBES		{0x0, 0x0}
+
 /* commands to include */
 #include <config_cmd_default.h>

-- 
1.5.6.3

--0016e64cc3d48ce7bb047ca8fca8
Content-Type: text/plain; charset=ISO-8859-1

-- 
You received this message because you are subscribed to the Google Groups "Beagle Board" group.
To post to this group, send email to beagleboard@googlegroups.com.
To unsubscribe from this group, send email to beagleboard+unsubscribe@googlegroups.com.
For more options, visit this group at http://groups.google.com/group/beagleboard?hl=en.



--0016e64cc3d48ce7bb047ca8fca8--
